src/codegen/compile.ml

```diff
@@ -3838,25 +3853,20 @@ module StableMem = struct
     match E.mode env with
     | Flags.ICMode | Flags.RefMode ->
       Func.share_code2 env "__stablemem_guard_range"
-        (("offset", I32Type), ("size", I32Type)) []
+        (("offset", I64Type), ("size", I32Type)) []
```

Comment:
Should size not be i64 as well?

Reply:
Hmm, I wonder why the tests didn't fail - this should probably produce illegal code

Reply:
Oh, no its fine - I only ever use this with i32 sizes